### PR TITLE
PWN-7381 - Add new receive solana fragment

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7381
]
| Screenshot #1 | Screenshot #2 | Screenshot #3 |
|:-------------:|:-------------:|:-------------:|
| [Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3] |
Screenshot 2023-03-07 at 19 11 41

### CODE REVIEW

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaFragment.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+private const val ARG_TOKEN_SYMBOL = "ARG_TOKEN_SYMBOL"
+
+class NewReceiveSolanaFragment :
+    BaseMvpFragment<NewReceiveSolanaContract.View, NewReceiveSolanaContract.Presenter>(
+        R.layout.fragment_new_receive_solana
+    ),
+    NewReceiveSolanaContract.View {
+
+    companion object {
+        fun create(address: String, logoUrl: String, tokenSymbol: String) = NewReceiveSolanaFragment()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaFragment.kt:29]: address -> solAddress

File: app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaFragment.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    private val tokenAddress: String by args(ARG_SOL_ADDRESS)
+    private val logoUrl: String by args(ARG_TOKEN_LOGO_URL)
+    private val tokenSymbol: String by args(ARG_TOKEN_SYMBOL)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            textViewAddress.text = tokenAddress
+            glideManager.load(watermarkImageView, logoUrl)
+            toolbar.title = getString(R.string.receive_on_solana, tokenSymbol)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaFragment.kt:49]: firstly you init toolbar and after that other views
it's just common sense - you init views in order that they are on screen

File: app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaFragment.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        binding.qrImageView.setImageBitmap(qrBitmap)
+        showUsername(username)
+    }
+
+    override fun showLoading(isLoading: Boolean) {
+        binding.progressView.isVisible = isLoading
+    }
+
+    private fun showUsername(username: String) {
+        if (username.isEmpty()) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaFragment.kt:80]: better make it nullable
if there no username (username == null) render view this way, or other way

File: app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            try {
+                view?.showLoading(isLoading = true)
+                val qr = qrCodeInteractor.generateQrCode(tokenAddress)
+                val username = usernameInteractor.getUsername()?.fullUsername.orEmpty()
+                view?.showQrAndUsername(qr, username)
+            } catch (e: CancellationException) {
+                Timber.d("Qr generation was cancelled")
+            } catch (e: Throwable) {
+                Timber.e(e, "Failed to generate qr bitmap")
+                view?.showErrorMessage()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/receive/solana/NewReceiveSolanaPresenter.kt:27]: no error message passed inside

File: app/src/main/res/layout/fragment_new_receive_solana.xml [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    <org.p2p.uikit.organisms.UiKitToolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toTopOf="parent"
+        app:navigationIcon="@drawable/ic_back"
+        app:title="@string/receive_on_solana" />
+
+    <FrameLayout
+        android:id="@+id/qrFrame"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_receive_solana.xml:19]: containerQr

File: app/src/main/res/layout/fragment_new_receive_solana.xml [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+        <FrameLayout
+            android:id="@+id/watermarkImageViewBackground"
+            android:layout_width="36dp"
+            android:layout_height="36dp"
+            android:layout_gravity="center"
+            android:background="@drawable/bg_circle">
+
+            <org.p2p.uikit.atoms.UiKitRoundedImageView
+                android:id="@+id/watermarkImageView"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_receive_solana.xml:44]: imageViewWatermark

File: app/src/main/res/layout/fragment_new_receive_solana.xml [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                android:layout_gravity="center"
+                android:src="@drawable/ic_sol"
+                app:cornerRadius="16dp" />
+
+        </FrameLayout>
+
+    </FrameLayout>
+
+    <LinearLayout
+        android:id="@+id/layoutAddress"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_receive_solana.xml:56]: containerAddress

layout is used for <include />

File: app/src/main/res/layout/fragment_new_receive_solana.xml [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:textColor="@color/text_mountain"
+            app:layout_constraintTop_toBottomOf="@id/textViewUsernameTitle"
+            tools:text="username.key" />
+
+
+    </LinearLayout>
+
+
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_receive_solana.xml:126]:
```suggestion
```

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**